x/ecocredit/core/msg_credit_type.go

```diff
@@ -0,0 +1,28 @@
+package core
+
+import (
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+const (
+	maxCreditTypeNameLength = 75
+)
+
+func (m CreditType) Validate() error {
+	if err := ValidateCreditTypeAbbreviation(m.Abbreviation); err != nil {
```

Comment:
it appears this function is duplicated in our code base

Reply:
mostly just leftovers from params right? should get removed with https://github.com/regen-network/regen-ledger/issues/1016

Reply:
I think aaron is referring to the error in the ci checks:

```
CreditType.Validate redeclared in this block
```

https://github.com/regen-network/regen-ledger/runs/6085436937?check_suite_focus=true

Reply:
ah got it -- [fixed](https://github.com/regen-network/regen-ledger/pull/1015/commits/eb1b2f56003f1b7cfc0edf93b27f2674d5ab2009)